Description: Disable assembler in WebKit on unsupported architectures
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
Taken from: https://sources.debian.org/patches/openjfx/11.0.2+1-1/07-disable-assembler-on-unsupported-archs.patch
--- old/modules/javafx.web/src/main/native/Source/WTF/wtf/Platform.h
+++ new/modules/javafx.web/src/main/native/Source/WTF/wtf/Platform.h
@@ -357,6 +357,12 @@
 #endif
 #endif
 
+#if !CPU(ARM_THUMB2) && !CPU(ARM_TRADITIONAL) && !CPU(X86) && !CPU(X86_64) && !CPU(SH4)
+#define ENABLE_ASSEMBLER 0
+#define ENABLE_JIT 0
+#define ENABLE_YARR_JIT 0
+#endif
+
 /* ==== OS() - underlying operating system; only to be used for mandated low-level services like
    virtual memory, not to choose a GUI toolkit ==== */
 
